checker/src/main/java/org/checkerframework/checker/signature/qual/SourceNameForNonArrayNonInner.java

```diff
@@ -14,6 +15,7 @@
  *
  * @checker_framework.manual #signature-checker Signature Checker
  */
-@SubtypeOf({SourceNameForNonInner.class, BinaryNameForNonArray.class})
+@SubtypeOf({FullyQualifiedName.class, BinaryName.class})
+@ImplicitFor(stringPatterns = "^[A-Za-z_][A-Za-z_0-9]*(\\.[A-Za-z_][A-Za-z_0-9]*)*$")
 @Target({ElementType.TYPE_USE, ElementType.TYPE_PARAMETER})
 public @interface SourceNameForNonArrayNonInner {}
```

Comment:
This is an ugly name, and one that isn't intuitive to users.  (That isn't your fault!  The name predates your work.)
Let's rename this to `DotSeparatedIdentifiers`.  Don't forget to update the Javadoc too.  Thanks.